app/lib/common/dialogs/member_info_drawer.dart

```diff
@@ -0,0 +1,264 @@
+            ),
+          ),
+          child: ActerAvatar(
+            mode: DisplayMode.DM,
+            avatarInfo: AvatarInfo(
+              uniqueId: memberId,
+              avatar: memberProfile.getAvatarImage(),
+              displayName: memberProfile.displayName,
+            ),
+            size: 50,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildMessageUser() {
+    return Consumer(
+      builder: (context, ref, widget) {
+        final client = ref.watch(alwaysClientProvider);
+        final dmId = client.dmWithUser(memberId).text();
+        if (dmId != null) {
+          debugPrint('DM id; $dmId');
+          return Center(
+            child: OutlinedButton.icon(
+              icon: const Icon(Atlas.chats_thin),
+              onPressed: () async {
+                context.pop();
+                context.pushNamed(
+                  Routes.chatroom.name,
+                  pathParameters: {
+                    'roomId': dmId,
+                  },
+                );
+              },
+              label: const Text('Message'),
+            ),
+          );
+        } else {
+          return Center(
+            child: OutlinedButton.icon(
+              icon: const Icon(Atlas.chats_thin),
+              onPressed: () {
+                final profile = member.getProfile();
+                ref.read(createChatSelectedUsersProvider.notifier).state = [
+                  profile,
+                ];
+                context.pop();
+                context.pushNamed(
+                  Routes.createChat.name,
+                );
+              },
+              label: const Text('Start DM'),
+            ),
+          );
+        }
+      },
+    );
+  }
+
+  Widget _buildDisplayName(BuildContext context) {
+    return GestureDetector(
+      onTap: () async {
+        await gotoFullProfile(context);
+      },
+      child: Center(
+        child: Text(memberProfile.displayName!), // FIXME: make this prettier
+      ),
+    );
+  }
+
+  Widget _buildUserName(BuildContext context) {
+    return GestureDetector(
+      onTap: () async {
+        context.pop(); // close the drawer
+        Clipboard.setData(
+          ClipboardData(
+            text: memberId,
+          ),
+        );
+        customMsgSnackbar(
+          context,
+          'Username copied to clipboard',
+        );
+      },
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Text(memberId), // FIXME: make this prettier
+          const SizedBox(width: 5),
+          const Icon(Icons.copy_outlined),
+        ],
+      ),
+    );
+  }
+
+  Future<void> gotoFullProfile(BuildContext context) async {}
+}
+
+Widget _actionMenuItem({
+  Key? key,
+  required BuildContext context,
+  required IconData iconData,
+  Color? iconColor,
+  required String title,
+  TextStyle? titleStyles,
+  String? subTitle,
+  VoidCallback? onTap,
+  bool enable = true,
+  bool withMenu = true,
+}) {
```

Comment:
Do we really required all this params here?

Reply:
Yes I have created this type of reusable component. But it was totally based on the settings page item requirements. So that's why I putted to settings page level only instead of making it project level component.

In `member_info_drawer.dart`, our user-case it pretty straight forward we can make manage directly with default Flutter `ListTile` widget.